src/sources/util/body_decoding.rs

```diff
@@ -62,6 +63,9 @@ pub fn decode_body(body: Bytes, enc: Encoding) -> Result<Vec<Event>, ErrorMessag
                 .map_err(|error| json_error(format!("Error parsing Json: {:?}", error)))?;
             json_parse_array_of_object(parsed_json)
         }
+        Encoding::Binary => std::iter::once(Ok(body))
+            .map(|r| Ok(LogEvent::from(r?).into()))
+            .collect::<Result<_, _>>(),
```

Comment:
This looks like a complicated way of producing the result. Is this anything more complicated than the following?
```rust
Ok(vec![LogEvent::from(body).into()])
```

Reply:
Thanks for simplification, that was basically stolen from text encoding part, already simplified according yours suggestion.